shared/filestore/s3store.go

```diff
@@ -157,16 +158,18 @@ func (b *S3FileBackend) TestConnection() error {
 		}
 	}
 
-	if exists {
-		mlog.Debug("Connection to S3 or minio is good. Bucket exists.")
-	} else {
-		mlog.Warn("Bucket specified does not exist. Attempting to create...")
-		err := b.client.MakeBucket(context.Background(), b.bucket, s3.MakeBucketOptions{Region: b.region})
-		if err != nil {
-			return errors.Wrap(err, "unable to create the s3 bucket")
-		}
+	if !exists {
+		return ErrNoS3Bucket
 	}
+	mlog.Debug("Connection to S3 or minio is good. Bucket exists.")
+	return nil
+}
 
+func (b *S3FileBackend) MakeBucket() error {
```

Comment:
Can we write tests for this function? One where it succeeds and one where it fails (mock the fail).

Reply:
This function is fairly short - what kind of test did you have in mind?

Reply:
See if you can mock the error and make sure it returns that. Although pretty sure, just wanna make sure we have some test coverage for any new function introduced even if something simple.